worker/worker/ffmegtools.go

```diff
@@ -14,12 +16,29 @@ func getDuration(file string) (float64, error) {
 	return gjson.Get(probe, "format.duration").Float(), nil
 }
 
-func getCodec(file string) (string, error) {
+func getCodec(file string, codecType string) (string, error) {
 	probe, err := probe(file)
 	if err != nil {
 		return "", err
 	}
-	return gjson.Get(probe, "streams.0.codec_name").String(), nil
+	nStreams := gjson.Get(probe, "streams.#").Int()
+	if codecType == "video" {
+		for i := 0; i < int(nStreams); i++ {
+			if gjson.Get(probe, fmt.Sprintf("streams.%d.codec_type", i)).String() == "video" {
+				return gjson.Get(probe, fmt.Sprintf("streams.%d.codec_name", i)).String(), nil
+			}
+		}
+		return "", errors.New("no video stream found")
+	}
+	if codecType == "audio" {
+		for i := 0; i < int(nStreams); i++ {
+			if gjson.Get(probe, fmt.Sprintf("streams.%d.codec_type", i)).String() == "audio" {
+				return gjson.Get(probe, fmt.Sprintf("streams.%d.codec_name", i)).String(), nil
+			}
+		}
+		return "", errors.New("no audio stream found")
```

Comment:
This should do the exact same, no?

```suggestion
	for i := 0; i < int(nStreams); i++ {
		if gjson.Get(probe, fmt.Sprintf("streams.%d.codec_type", i)).String() == codecType {
			return gjson.Get(probe, fmt.Sprintf("streams.%d.codec_name", i)).String(), nil
		}
	}
	return "", fmt.Errorf("no %s stream found", streamType)
```

Reply:
Yeah you are right, this is much more compact than my version. I will push this change before today's meeting.